src/sentry/tagstore/legacy/backend.py

```diff
@@ -56,7 +59,13 @@ def setup(self):
         ])
         default_manager.add_bulk_dependencies(Event, [
             lambda instance_list: ModelRelation(EventTag,
-                                                {'event_id__in': [i.id for i in instance_list]},
+                                                {
+                                                    'event_id__in': [i.id for i in instance_list],
+                                                    'project_id__in': {
```

Comment:
Can't do `IN` across shards either. Deletion code will have to be refactored to allow for one query per `project_id`. Which sucks, because I think when we delete groups we're just iterating by date? So the project constantly varies. Will be a large refactor I think to get that correct/fast.

Other option: https://docs.citusdata.com/en/v7.0/reference/user_defined_functions.html#master-modify-multiple-shards